broker-daemon/index.spec.js

```diff
@@ -414,6 +406,60 @@ describe('broker daemon', () => {
     })
   })
 
+  describe('#validateEngines', () => {
+    let attempts
+    let delay
+    let btcEngine
+    let ltcEngine
+
+    beforeEach(() => {
+      brokerDaemon = new BrokerDaemon(brokerDaemonOptions)
+
+      btcEngine = {
+        validateNodeConfig: sinon.stub().resolves()
+      }
+      ltcEngine = {
+        validateNodeConfig: sinon.stub().resolves()
+      }
+      brokerDaemon.engines = new Map([
+        [ 'BTC', btcEngine ],
+        [ 'LTC', ltcEngine ]
+      ])
+
+      attempts = BrokerDaemon.__get__('EXPONENTIAL_BACKOFF_ATTEMPTS')
+      delay = BrokerDaemon.__get__('EXPONENTIAL_BACKOFF_DELAY')
+    })
+
+    it('passes engine validation to exponential backoff for each engine', () => {
+      brokerDaemon.validateEngines()
+
+      const validateBtcEngine = exponentialBackoff.args[0][0]
+      validateBtcEngine()
+
+      expect(btcEngine.validateNodeConfig).to.have.been.called()
+
+      const validateLtcEngine = exponentialBackoff.args[1][0]
+      validateLtcEngine()
+
+      expect(ltcEngine.validateNodeConfig).to.have.been.called()
+
+      expect(exponentialBackoff).to.have.been.calledTwice()
+      expect(exponentialBackoff).to.have.been.calledWith(
+        sinon.match.func,
```

Comment:
you should add some tests for the function that gets passed to `exponentialBackoff` to make sure that it does what you expect it to (i.e. that it calls `validateNodeConfig`)

Reply:
I see that you actually did this - maybe just split it up into different tests so that it is clear?